django/db/models/lookups.py

```diff
@@ -367,9 +367,9 @@ def process_rhs(self, compiler, connection):
 
         if self.rhs_is_direct_value():
             try:
-                rhs = OrderedSet(self.rhs)
+                rhs = OrderedSet(r for r in self.rhs if r is not None)
```

Comment:
We could add a line of comment because this [observation](https://code.djangoproject.com/ticket/31667#comment:1) may not obvious to everyone.

Reply:
Aye that is faster

```
In [6]: %timeit items = [None] + list(range(100)) ; items = OrderedSet(i for i in items if i is not None)
8.71 µs ± 64.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [7]: %timeit items = [None] + list(range(100)) ; items = OrderedSet(items) ; items.discard(None)
4.89 µs ± 23.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
What about a [small comment](https://github.com/django/django/pull/13030#discussion_r437941823)? :wink:

Reply:
Oh yes, added.